src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt

```diff
@@ -106,7 +106,9 @@ class ActivitiesService(
         .also { telemetryMap["dpsActivityId"] = it.id.toString() }
 
       activitySchedule.toUpdateActivityRequest(activity.pay, activity.category.code)
-        .also { nomisApiService.updateActivity(nomisCourseActivityId, it) }
+        .let { nomisApiService.updateActivity(nomisCourseActivityId, it) }
+        .let { nomisResponse -> buildActivityMappingDto(nomisResponse, activitySchedule) }
+        .also { mappingRequest -> mappingService.updateMapping(mappingRequest) }
```

Comment:
The main change - we now maintain the schedule mappings returned when updating an Activity in case any schedules were created / deleted.